include/__generator.hpp

```diff
@@ -516,6 +516,18 @@ struct __generator_promise<generator<_Ref, _Value, _Alloc>, _ByteAllocator, _Exp
     }
 };
 
+template<typename _Alloc>
+struct __byte_allocator {
+    using type = typename std::allocator_traits<std::remove_cvref_t<_Alloc>>::template rebind_alloc<std::byte>;
+};
+template<>
+struct __byte_allocator<std::use_allocator_arg>  {
+    using type = std::allocator<std::byte>;
+};
```

Comment:
I don't think we should be defining this specialisation here as it now gives meaning to `std::generator<int> foo(std::allocator_arg_t, std::use_allocator_arg)` which is kind of nonsensical.

Go for adding a separate overload of `yield_value(ranges::elements_of<Rng, use_allocator_arg>)` to handle the stateless allocator case instead.